client/state/posts/likes/reducer.js

```diff
@@ -104,7 +104,7 @@ export const itemReducer = withSchemaValidation(
 
 				let likes = state.likes;
 				if ( hasLiker ) {
-					likes = dropWhile( state.likes, ( l ) => liker.ID === l.ID );
```

Comment:
What do we want the code to do here? Simply remove all items from the `likes` array that have a matching `ID`? That's what `filter` does, yes, but `dropWhile` did something very different.

`dropWhile` is something like `trimStart` with strings and leading spaces: it removes matching elements only from the beginning:
```js
dropWhile( [ 1, 1, 2, 1 ], i => i === 1 ) === [ 2, 1 ]; // stops at `2`
dropWhile( [ 2, 1, 2, 1 ], i => i === 1 ) === [ 2, 1, 2, 1 ]; // no `1` at the beginning
```

Does that mean that the code was buggy until now and didn't remove likers if they are in the middle of the list?

The unit tests only test cases where there's only one liker, never with multiple ones. So they don't catch this.

Reply:
Thanks for pointing it out. While working on it I did notice the behavior and totally believe it's supposed to just remove the matching element and leave all others. That's what I'd expect from a removal reducer, after all. `dropWhile` had failed to do this properly, while `filter` does it well. 

I've expanded the test to account for this case in f84cc2973afe29e8916ac53e7071be20d2a72578 and will merge unless you have objections.